src/main/java/school/hei/haapi/service/event/UpdateStudentsStatusService.java

```diff
@@ -0,0 +1,38 @@
+package school.hei.haapi.service.event;
+
+import java.util.List;
+import java.util.function.Consumer;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+import school.hei.haapi.endpoint.event.model.UpdateStudentsStatus;
+import school.hei.haapi.model.User;
+import school.hei.haapi.service.PaymentService;
+import school.hei.haapi.service.UserService;
+
+@Service
+@AllArgsConstructor
+public class UpdateStudentsStatusService implements Consumer<UpdateStudentsStatus> {
+
+  private final UserService userService;
+  private final PaymentService paymentService;
+
+  public void checkSuspendedStudentsStatus() {
+    List<User> suspendedStudents = userService.getAllSuspendedUsers();
+    for (User student : suspendedStudents) {
+      paymentService.computeUserStatusAfterPayingFee(student);
+    }
+  }
+
+  public void checkEnabledStudentsStatus() {
+    List<User> enabledStudents = userService.getAllEnabledUsers();
```

Comment:
I think we can try something better. I answer my self that, 90% and more of STUDENT are ENABLED so computing something for 90% of user or 300 STUDENTS it's very hard and heavy